### PR TITLE
fix: Fix exports, add depTree to exports too

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
- expose top level types without the need to reference `dist`
- export `depTree` type

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
-        "es2015",
+        "es2015"
       ],
       "module": "commonjs",
       "allowJs": false,
       "sourceMap": true,
       "strict": true,
       "declaration": true,
       "importHelpers": true,
-      "noImplicitAny": false // Needed to compile tap and ansi-escapes
```
Comment (miiila) [inline tsconfig.json]: I would keep this comment...

File: lib/legacy/index.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import * as pluginInterface from './plugin';
-import * as monitorInterface from './monitor';
+import * as plugin from './plugin';
+import * as monitor from './monitor';
+import * as common from './monitor';
 
-export {pluginInterface, monitorInterface};
```
Comment (miiila) [inline lib/legacy/index.ts]: TBH, this is technically breaking change. Also, I don't understand how naming change helps with `the need to reference dist`.

Comment (lili2311) [reply lib/legacy/index.ts]: we don't want to reference dist, so we want to structure our exports in a way we want to import them. Given how we use `dist` it seems we want to `types.legacy.plugin.Blah`
without it we will need to go `types.legacy. pluginInterface.Blah` agreed it needs a chat :)

Comment (lili2311) [reply lib/legacy/index.ts]: @kyegupov @gjvis @snyk/boost

File: tsconfig.json (hunk 1 above)
Comment (kyegupov) [reply tsconfig.json]: it's a lie now, I think.

File: lib/legacy/index.ts (hunk 2 above)
Comment (kyegupov) [reply lib/legacy/index.ts]: With this, I still cannot `import {Blah} from '@snyk/cli-interface/types/legacy/plugin'`.

**Pull request closed without merging**